shopinvader/templates/deployment.yaml

```diff
@@ -54,6 +54,17 @@ spec:
           ports:
             - name: shopinvader
               containerPort: 3000
+          env:
+            - name: S3_KEY_ID
+              valueFrom:
+                secretKeyRef:
+                  name: s3-gateway-minio
```

Comment:
you need to use templating for this, something like: `shopinvader.secrets.s3.accessKey.secretName`, `shopinvader.secrets.s3.accessKey.secretKey` and same for secret key.
Then you can use `s3-gateway-minio` as a default value

Reply:
you can also use a standard additional env pattern